src/main/java/codeu/controller/ChatServlet.java

```diff
@@ -142,14 +167,44 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
 
     String messageContent = request.getParameter("message");
 
+    UUID messageUUID = UUID.randomUUID();
+
     // this removes any HTML from the message content
     String cleanedMessageContent = Jsoup.clean(messageContent, Whitelist.none());
     
     String cleanedAndEmojiMessage = EmojiParser.parseToUnicode(cleanedMessageContent);
 
+    Pattern mentionPattern = Pattern.compile("@[^@]+(\\s|\\n|$)");
+
+    Matcher mentionMatch = mentionPattern.matcher(cleanedAndEmojiMessage);
+
+
+    Set<String> mentionedUsers = new HashSet<String>();
+
+    while (mentionMatch.find()) {
+      String mentionedUser = mentionMatch.group();
+      mentionedUser = mentionedUser.trim();
+      mentionedUser = mentionedUser.substring(1);
```

Comment:
I'd combine this into
```java
String mentionedUser = mentionMatch.group().trim().substring(1);
```

As a general rule of thumb if I'm just manipulating the same variable I'll just condense it to a single line